etna/search/tests/test_views.py

```diff
@@ -223,6 +223,361 @@ def test_accessing_page_with_no_params_performs_empty_search(self):
+        """
+        self.patch_search_endpoint("website_search_no_results.json")
+        response = self.client.get(self.test_url, data={"q": "qwerty"})
+        content = str(response.content)
+
+        # SHOULD see
+        self.assertNoResultsMessagingRendered(content)
+
+        # SHOULD NOT see
+        self.assertBucketLinksNotRendered(content)
+        self.assertSearchWithinOptionNotRendered(content)
+        self.assertSortOrderOptionsNotRendered(content)
+        self.assertFilterOptionsNotRendered(content)
+        self.assertResultsNotRendered(content)
+
+    @responses.activate
+    def test_no_matches_for_the_current_bucket(self):
+        """
+        When a user searches for something that has results for SOME buckets,
+        but they are currently viewing a bucket with no results:
+
+        They SHOULD see:
+        - Names, result counts and links for all buckets
+        - A "No results" message.
+
+        They SHOULD NOT see:
+        - A "Search within these results" option
+        - Options to change sort order and display style of results
+        - Filter options to refine the search
+        - Search results
+        """
+        self.patch_search_endpoint("website_search_with_some_empty_buckets.json")
+        response = self.client.get(self.test_url, data={"q": "japan", "group": "audio"})
+        content = str(response.content)
+
+        # SHOULD see
+        self.assertBucketLinksRendered(content)
+        self.assertNoResultsMessagingRendered(content)
+
+        # SHOULD NOT see
+        self.assertSearchWithinOptionNotRendered(content)
+        self.assertSortOrderOptionsNotRendered(content)
+        self.assertFilterOptionsNotRendered(content)
+        self.assertResultsNotRendered(content)
+
+    @responses.activate
+    def test_no_matches_for_the_refined_search(self):
+        """
+        When a user is viewing a bucket with results for the original search,
+        then uses the "Search within these results" option to refine it, and
+        that 'refined' search has no results:
+
+        They SHOULD see:
+        - Names, result counts and links for all buckets
+        - A "Search within these results" option
+        - Options to change sort order and display style of results
+        - Filter options to refine the search
+        - A "No results" message.
+
+        They SHOULD NOT see:
+        - Search results
+        """
+
+        self.patch_search_endpoint("website_search_empty_after_refining.json")
+        response = self.client.get(
+            self.test_url,
+            data={"q": "japan", "group": "blog", "filter_keyword": "qwerty"},
+        )
+        content = str(response.content)
+
+        # SHOULD see
+        self.assertBucketLinksRendered(content)
+        self.assertSearchWithinOptionRendered(content)
+        self.assertSortOrderOptionsRendered(content)
+        self.assertFilterOptionsRendered(content)
+        self.assertNoResultsMessagingRendered(content)
+
```

Comment:
The text in `no_results_messaging_html` ..._We did not find any results for your search_ is common across templates\search\blocks\no_results.html and templates\search\blocks\no_results_catalogue_website.html

Could we then have one which is specific for a Refined search with text "_Try removing any filters that you may have applied_" from `no_results_catalogue_website.html`

Reply:
I'm trying to deliberately steer away from testing actual messaging, as that is not really a testing concern, but an editorial one.

Reply:
This is great. Thanks @ababic